automate/scripts/etl_utils.py

```diff
@@ -0,0 +1,440 @@
+import argparse
+import docker
+import subprocess
+import glob
+import os
+import boto3
+import zipfile
+import shutil
+import etl_constants
+import psycopg2
+import json
+import time
+import requests
+from fhir.resources.R4B.fhirtypesvalidators import bundle_validator
+from pathlib import Path
+    
+def run_docker_compose(yaml_path):
+    try:
+        # Run Docker Compose command in CMD
+        with subprocess.Popen(f'docker-compose -f "{yaml_path}" up -d', shell=True) as process:
+            process.wait()  # Wait for the process to complete
+        print("Docker Compose executed successfully.")
+    except subprocess.CalledProcessError as e:
+        print("Error running Docker Compose command:", e)
+        
+def run_docker_compose_with_logs(yaml_path):
+    try:
+        # Run Docker Compose command in CMD
+        with subprocess.Popen(f'docker-compose -f "{yaml_path}" up', shell=True) as process:
+            process.wait()  # Wait for the process to complete
+        print("Docker Compose executed successfully.")
+    except subprocess.CalledProcessError as e:
+        print("Error running Docker Compose command:", e)
+
+def remove_docker_containers(yaml_path):
+    try:
+        # Run Docker Compose command in CMD
+        with subprocess.Popen(f'docker-compose -f "{yaml_path}" down', shell=True) as process:
+            process.wait()  # Wait for the process to complete
+        print("Docker Compose down executed successfully.")
+    except subprocess.CalledProcessError as e:
+        print("Error running Docker Compose command:", e)
+
+def remove_docker_volume(volume_name):
+    client = docker.from_env()
+    try:
+        volume = client.volumes.get(volume_name)
+        if volume:
+            volume.remove()
+            print(f"Docker volume '{volume_name}' removed successfully.")
+        else:
+            print(f"Error: Docker volume '{volume_name}' not found.")
+    except docker.errors.APIError as e:
+        print(f"Error removing Docker volume '{volume_name}': {e}")
+    # try:
+    #     # Run Docker volume removal command in CMD
+    #     subprocess.run(f'docker volume rm "{volume_name}"', check=True)
+    #     print("Docker volume removed successfully.")
+    # except subprocess.CalledProcessError as e:
+    #     print("Error removing Docker volume:", e)
+
+def validate_file_for_fhir(file_path):
+    try: 
+        # Assuming bundle_data contains the Bundle resource data in dictionary format
+        bundle_data = Path(file_path)
+
+        # Validate the Bundle resource
+        bundle_validator(bundle_data)
+        print("Bundle Validated Successfully")
+        return True
+    except ValueError as e:
+        print("Validation errors:")
+        print(e)
+        return False
+
+    except Exception as e:
+        print(f"An error occurred: {e}")
```

Comment:
Suggestion: If you want to print the stack trace then use `traceback.print_exc()`